app/models/user.rb

```diff
@@ -86,6 +86,27 @@ class User < ActiveRecord::Base
     },
     ranked_by: ":tsearch + (0.25 * :trigram)"
 
+  def self.dormant(window=5)
+    DatabaseReplica.read("read_dormant_users_from_replica") do
+      # devise trackable sets the current_sign_in_at on each login
+      havent_signed_in_since = "date(now()) - date(current_sign_in_at) >= #{window}"
+      where(havent_signed_in_since).find_each do |dormant_user|
+        # Check if dormant user has classified in last 5 days, only yield block if meet warden_conditions
+        latest_user_classification = Classification.where(user_id: dormant_user).order(created_at: :desc).first
```

Comment:
Also i'm wondering if we can use the `Recent` class (table) for this query instead of the `Classification`, it might be a smaller index / table space search? 

I am happy to help test this out too.